diff -u -r libdsm-0.2.8/src/smb_utils.c unxlngr/libdsm-0.2.8/src/smb_utils.c
--- libdsm-0.2.8/src/smb_utils.c	2019-01-16 14:18:11.000000000 +0900
+++ unxlngr/libdsm-0.2.8/src/smb_utils.c	2019-01-16 14:19:35.000000000 +0900
@@ -40,7 +40,7 @@
 #include <string.h>
 #include <errno.h>
 
-#if HAVE_LANGINFO_H && !defined( __APPLE__ )
+#if HAVE_LANGINFO_H && !defined( __APPLE__ ) && !defined(ANDROID)
 # include <langinfo.h>
 #endif
 
@@ -51,7 +51,7 @@
 {
 #if defined( __APPLE__ )
     return "UTF8";
-#elif !HAVE_LANGINFO_H
+#elif !HAVE_LANGINFO_H || defined(ANDROID)
     return "UTF-8";
 #else
     static int locale_set = 0;
